### PR TITLE
Add support for Vault namespaces (used in Vault Enterprise)

### LINKED ISSUE #38: Add support for Vault Namespace
Hi, 

It would be great if `helm-vault` provided support for [Vault Namespaces](https://www.vaultproject.io/docs/enterprise/namespaces) found in Vault Enterprise. For example this can be implemented by using the `VAULT_NAMESPACE` environment variable. A similar functionality can be found in `vault-cli` in which the namespace can be configured using this environment variable.

Cheers,
Marius

(PR follows)

### PR DESCRIPTION
**Pull request by mariusgiger**
# Add support for Vault namespaces (used in Vault Enterprise)

## Description

This PR adds support Vault Enterprise namespaces. In accordance with `vault-cli` the Vault Namespace can be configured using the environment variable `VAULT_NAMESPACE`.

Also refer to: https://www.vaultproject.io/docs/commands#vault_namespace

Fixes #38 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

## Mentions:

@Just-Insane

### CODE REVIEW

File: src/vault.py [hunk 1]
```diff
@@ -198,7 +198,7 @@ def __init__(self, args, envs):
 
         # Setup Vault client (hvac)
         try:
-            self.client = hvac.Client(url=self.envs.vault_addr, token=os.environ["VAULT_TOKEN"])
+            self.client = hvac.Client(url=self.envs.vault_addr, namespace=os.environ["VAULT_NAMESPACE"], token=os.environ["VAULT_TOKEN"])
```
Comment (Just-Insane) [inline src/vault.py]: It appears VAULT_NAMESPACE becomes mandatory when set this way, can it be made optional, and an additional error message added?

**Comment by Just-Insane** [comment]
Hello,

It appears this fails if VAULT_NAMESPACE is not provided, can this be made optional?

Some additional tests around vault namespaces would be needed as well.

**Comment by mariusgiger** [comment]
> Hello,
> 
> It appears this fails if VAULT_NAMESPACE is not provided, can this be made optional?
> 
> Some additional tests around vault namespaces would be needed as well.

Hi @Just-Insane, thanks for the feedback. I have fixed the issue where VAULT_NAMESPACE was required and have added some scripts to allow running tests locally without setting up a vault beforehand. Let me know what you think. When it comes to additional tests I am not sure how to best test this as this is an enterprise feature. Any hints?

Cheers, 
Marius

**Comment by Just-Insane** [comment]
I'm also not sure how to test the Vault namespace feature, as I don't have an enterprise license.

It looks like this is all part of `hvac` so I'm not too worried about testing it within helm_vault itself.

**Pull request merged**